### PR TITLE
Updated workflow to run mock tests then external tests

### PR DESCRIPTION
**Pull request by Mattsface**
### Why
As a developer I want mock tests and external tests to be different actions in a pipeline

### What
Mock tests and external tests are separate

### Tests
PyTest

### Risk and impact
- Minimal

### CODE REVIEW

File: tests/external_tests/gamepace/test_gamepace.py [hunk 1]
```diff
@@ -3,65 +3,65 @@
 from mlbstatsapi import Mlb
 
 
-class TestGamepace(unittest.TestCase):
-    @classmethod
-    def setUpClass(cls) -> None:
-        cls.mlb = Mlb()
+# class TestGamepace(unittest.TestCase):
```
Comment (KCNilssen) [inline tests/external_tests/gamepace/test_gamepace.py:6]: Is gamepace still causing issues?

Comment (Mattsface) [reply tests/external_tests/gamepace/test_gamepace.py:6]: It seems to fail at random times that end point sucks.

Comment (KCNilssen) [reply tests/external_tests/gamepace/test_gamepace.py:6]: Cool, well keep i as a mock for now. That works for me until the mlb stats api can get their crap together and return a consistent result for the same query

**Review by KCNilssen — APPROVED** [review]
LGTM

**Pull request merged**